services/sharedchannel/getpostssince.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sharedchannel
+
+import (
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+type sinceResult struct {
+	posts     []*model.Post
+	hasMore   bool
+	nextSince int64
+}
+
+// getPostsSince fetches posts that need to be synchronized with a remote cluster.
+// There is a soft cap on the number of posts that will be synchronized in a single pass (MaxPostsPerSync).
+//
+// There is a special case where multiple posts have the same UpdateAt value. It is vital that this method
+// include all posts within that millisecond so that subsequent calls can use an incremented `since`. If this
+// method were to be called repeatedly with the same `since` value the same records would be returned each time
+// and the sync would never move forward.
+//
+// A boolean is also returned to indicate if there are more posts to be synchronized (true) or not (false).
+func (scs *Service) getPostsSince(channelId string, rc *model.RemoteCluster, since int64) (sinceResult, error) {
+	opts := model.GetPostsSinceForSyncOptions{
+		ChannelId: channelId,
+		Since:     since,
+		//ExcludeRemoteId: rc.RemoteId,
```

Comment:
Should this be removed?

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/16962/commits/3f4dc12f54db024764ab8b953b8d947174283d18